core/tests/protractor_desktop/creatorDashboard.js

```diff
@@ -0,0 +1,135 @@
+var waitFor = require('../protractor_utils/waitFor.js');
+var workflow = require('../protractor_utils/workflow.js');
+
+
+var AdminPage =
+require('../protractor_utils/AdminPage.js');
+var CreatorDashboardPage =
+require('../protractor_utils/CreatorDashboardPage.js');
+var CollectionEditorPage =
+require('../protractor_utils/CollectionEditorPage.js');
+var ExplorationEditorPage =
+require('../protractor_utils/ExplorationEditorPage.js');
+var ExplorationPlayerPage =
+require('../protractor_utils/ExplorationPlayerPage.js');
+var LibraryPage =
+require('../protractor_utils/LibraryPage.js');
+
+describe('Creator dashboard functionality', function() {
+  var creatorDashboardPage = null;
+  var explorationEditorPage = null;
+  var explorationPlayerPage = null;
+  var libraryPage = null;
+  var learnerDashboardPage = null;
+
+  beforeAll(function() {
+    libraryPage = new LibraryPage.LibraryPage();
+    collectionEditorPage = new CollectionEditorPage.CollectionEditorPage();
+    creatorDashboardPage = new CreatorDashboardPage.CreatorDashboardPage();
+    explorationEditorPage = new ExplorationEditorPage.ExplorationEditorPage();
+    explorationEditorMainTab = explorationEditorPage.getMainTab();
+    explorationEditorSettingsTab = explorationEditorPage.getSettingsTab();
+    explorationPlayerPage = new ExplorationPlayerPage.ExplorationPlayerPage();
+  });
+
+  it('displays creators subscribers', function() {
+    // created a creator ID and 2 learner ID.
+    var creator1Id = 'creatorName';
+    users.createUser('learner1@learnerDashboard.com',
+      'learner1learnerDashboard');
+    users.createUser('learner2@learnerDashboard.com',
+      'learner2learnerDashboard');
+    users.createUser(creator1Id + '@creatorDashboard.com', creator1Id);
+
+    // created a new exploration for the creator.
+    users.login(creator1Id + '@creatorDashboard.com');
+    creatorDashboardPage.clickCreateExplorationButton();
+    workflow.createAndPublishExploration(
+      'Activations',
+      'Chemistry',
+      'Learn about different types of chemistry activations.',
+      'English'
+    );
+    users.logout();
+
+    // both the learners will subscribe to the creator.
+    users.login('learner1@learnerDashboard.com');
+    subscriptionDashboardPage.navigateToUserSubscriptionPage(creator1Id);
+    subscriptionDashboardPage.navigateToSubscriptionButton();
+    users.logout();
+
+    users.login('learner2@learnerDashboard.com');
+    subscriptionDashboardPage.navigateToUserSubscriptionPage(creator1Id);
+    subscriptionDashboardPage.navigateToSubscriptionButton();
+    users.logout();
+
+    // exploration created by the creator.
+    users.login(creator1Id + '@creatorDashboard.com');
+    workflow.createAndPublishExploration(
+      'Activations',
+      'Chemistry',
+      'Learn about different types of chemistry activations.',
+      'English'
+    );
+
+    // navigating to creator's subscribed dashboard.
+    creatorDashboardPage.navigateToSubscriptionDashboard();
```

Comment:
Typically while writing tests, you perform an action, and then you assert that what you expect has happened. I don't see any assertions for your tests. Have a look at other files in protractor_desktop/ to get an idea.

Reply:
I tried exploring the creator dashboard page, and then i tried writing this test. All the actions of creating an exploration and publishing have been included in the test. What more actions should be added in this particular test?